File: pm_core_1/src/infrastructure/store/mapper/issue_type_scheme_item_mapper.go
package mapper

import (
	"github.com/serp/pm-core/src/core/domain/entity"
	"github.com/serp/pm-core/src/infrastructure/store/model"
)

type IssueTypeSchemeItemMapper struct{}

func NewIssueTypeSchemeItemMapper() *IssueTypeSchemeItemMapper {
	return &IssueTypeSchemeItemMapper{}
}

func (m *IssueTypeSchemeItemMapper) ToEntity(mdl *model.IssueTypeSchemeItemModel) *entity.IssueTypeSchemeItemEntity {
	if mdl == nil {
		return nil
	}
	return &entity.IssueTypeSchemeItemEntity{
		BaseEntity: entity.BaseEntity{
			ID:        mdl.ID,
			CreatedAt: mdl.CreatedAt.UnixMilli(),
			CreatedBy: mdl.CreatedBy,
			UpdatedAt: mdl.UpdatedAt.UnixMilli(),
			UpdatedBy: mdl.UpdatedBy,
		},
		TenantID:    mdl.TenantID,
		SchemeID:    mdl.SchemeID,
		IssueTypeID: mdl.IssueTypeID,
		Sequence:    mdl.Sequence,
	}
}

func (m *IssueTypeSchemeItemMapper) ToModel(e *entity.IssueTypeSchemeItemEntity) *model.IssueTypeSchemeItemModel {
	if e == nil {
		return nil
	}
	return &model.IssueTypeSchemeItemModel{
		BaseModel: model.BaseModel{
			ID:        e.ID,
			CreatedBy: e.CreatedBy,
			UpdatedBy: e.UpdatedBy,
		},
		TenantID:    e.TenantID,
		SchemeID:    e.SchemeID,
		IssueTypeID: e.IssueTypeID,
		Sequence:    e.Sequence,
	}
}

func (m *IssueTypeSchemeItemMapper) ToEntities(models []*model.IssueTypeSchemeItemModel) []*entity.IssueTypeSchemeItemEntity {
	if models == nil {
		return nil
	}
	entities := make([]*entity.IssueTypeSchemeItemEntity, 0, len(models))
	for _, mdl := range models {
		if e := m.ToEntity(mdl); e != nil {
			entities = append(entities, e)
		}
	}
	return entities
}

func (m *IssueTypeSchemeItemMapper) ToModels(entities []*entity.IssueTypeSchemeItemEntity) []*model.IssueTypeSchemeItemModel {
	if entities == nil {
		return nil
	}
	models := make([]*model.IssueTypeSchemeItemModel, 0, len(entities))
	for _, e := range entities {
		if mdl := m.ToModel(e); mdl != nil {
			models = append(models, mdl)
		}
	}
	return models
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/mapper/IssueTypeSchemeItemMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.mapper;

import org.springframework.stereotype.Component;
import serp.project.pmcore.core.domain.entity.IssueTypeSchemeItemEntity;
import serp.project.pmcore.infrastructure.store.model.IssueTypeSchemeItemModel;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class IssueTypeSchemeItemMapper extends BaseMapper {

    public IssueTypeSchemeItemEntity toEntity(IssueTypeSchemeItemModel model) {
        if (model == null) { return null; }
        return IssueTypeSchemeItemEntity.builder()
                .id(model.getId())
                .tenantId(model.getTenantId())
                .schemeId(model.getSchemeId())
                .issueTypeId(model.getIssueTypeId())
                .sequence(model.getSequence())
                .createdAt(localDateTimeToLong(model.getCreatedAt()))
                .createdBy(model.getCreatedBy())
                .updatedAt(localDateTimeToLong(model.getUpdatedAt()))
                .updatedBy(model.getUpdatedBy())
                .build();
    }

    public IssueTypeSchemeItemModel toModel(IssueTypeSchemeItemEntity entity) {
        if (entity == null) { return null; }
        return IssueTypeSchemeItemModel.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .schemeId(entity.getSchemeId())
                .issueTypeId(entity.getIssueTypeId())
                .sequence(entity.getSequence())
                .createdAt(longToLocalDateTime(entity.getCreatedAt()))
                .createdBy(entity.getCreatedBy())
                .updatedAt(longToLocalDateTime(entity.getUpdatedAt()))
                .updatedBy(entity.getUpdatedBy())
                .build();
    }

    public List<IssueTypeSchemeItemEntity> toEntities(List<IssueTypeSchemeItemModel> models) {
        if (models == null) { return Collections.emptyList(); }
        return models.stream().map(this::toEntity).collect(Collectors.toList());
    }

    public List<IssueTypeSchemeItemModel> toModels(List<IssueTypeSchemeItemEntity> entities) {
        if (entities == null) { return Collections.emptyList(); }
        return entities.stream().map(this::toModel).collect(Collectors.toList());
    }
}
